src/extractor/extraction_containers.cpp

```diff
@@ -567,41 +558,30 @@ void ExtractionContainers::WriteEdges(std::ofstream &file_out_stream) const
 
             // IMPORTANT: here, we're using slicing to only write the data from the base
             // class of NodeBasedEdgeWithOSM
-            NodeBasedEdge tmp = edge.result;
-            file_out_stream.write((char *)&tmp, sizeof(NodeBasedEdge));
-            used_edges_counter++;
+            normal_edges.push_back(edge.result);
```

Comment:
Alternatively we could use the approach of writing 0 as total number of elements, writing elements one by one, skipping back to the beginning and writing the appropriate count.